src/regressor.rs

```diff
@@ -352,6 +391,36 @@ impl Regressor {
         pb.observations.pop().unwrap()
     }
 
+    pub fn predict_with_cache(
+        &self,
+        fb: &feature_buffer::FeatureBuffer,
+        pb: &mut port_buffer::PortBuffer,
+        caches: &[BlockCache],
+    ) -> f32 {
+        pb.reset(); // empty the tape
+
+        let further_blocks = &self.blocks_boxes[..];
+        block_helpers::forward_with_cache(further_blocks, fb, pb, caches);
+
+        assert_eq!(pb.observations.len(), 1);
+        let prediction_probability = pb.observations.pop().unwrap();
+
+        return prediction_probability;
+    }
+
+    pub fn setup_cache(
```

Comment:
This implementation is general over all namespaces that appear right? Assuming one knows which parts of the input space/namespaces are actually the ones worth caching (e.g., top k namespaces) - wouldn't this speed this up even further as there would be less false positive things cached (the ones where lookups are not repeated over and over again?) (might have just missed that part)

Reply:
Indeed it is, but we also do not prepare those that do not contain features. For now, I implemented it so that we do not have major external API changes. 

As a follow-up, we could expand the command set to specify which namespaces should be used as part of the caching mechanism & when doing the preparation only compute those. The setup would also need to be expanded to include specific block operations (like with FFM use the correct indexes, etc).

Reply:
Agreed, it's not a non-intrusive change, thanks